src/chem/aeronet_aaod2ioda.py

```diff
@@ -133,83 +125,106 @@ def add_data(infile):
     outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-    loc_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-    var_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-    units = {}
-    units['latitude'] = 'degree'
-    units['longitude'] = 'degree'
-    units['station_elevation'] = 'm'
-
-    # Define AERONET inversion varname that match with those in f3
-    # (e.g., match aaeronetinv_wav and aeronetinv_chan)
-    obsvars = {'aerosol_optical_depth_3': 'aod_coincident_input[440nm]',
-               'aerosol_optical_depth_5': 'aod_coincident_input[675nm]',
-               'aerosol_optical_depth_6': 'aod_coincident_input[870nm]',
-               'aerosol_optical_depth_7': 'aod_coincident_input[1020nm]',
-               'absorption_aerosol_optical_depth_3': 'absorption_aod[440nm]',
-               'absorption_aerosol_optical_depth_5': 'absorption_aod[675nm]',
-               'absorption_aerosol_optical_depth_6': 'absorption_aod[870nm]',
-               'absorption_aerosol_optical_depth_7': 'absorption_aod[1020nm]'}
+    varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+
+    obsvars = {'aerosol_optical_depth': ['aod_coincident_input[440nm]', 'aod_coincident_input[675nm]',
+                                         'aod_coincident_input[870nm]', 'aod_coincident_input[1020nm]'],
+               'absorption_aerosol_optical_depth': ['absorption_aod[440nm]', 'absorption_aod[675nm]',
+                                                    'absorption_aod[870nm]', 'absorption_aod[1020nm]']}
+
+    AttrData = {
+        'converter': os.path.basename(__file__)
+    }
+
+    DimDict = {
+    }
+
+    VarDims = {
+        'aerosol_optical_depth': ['nlocs', 'nchans'],
+        'absorption_aerosol_optical_depth': ['nlocs', 'nchans'],
+        'frequency': ['nchans'],
+        'sensor_channel': ['nchans']
+    }
 
     # Define varDict variables
     for key, value in obsvars.items():
-        varDict[key]['valKey'] = key, writer.OvalName()
-        varDict[key]['errKey'] = key, writer.OerrName()
-        varDict[key]['qcKey'] = key, writer.OqcName()
+        varDict[key]['valKey'] = key, iconv.OvalName()
+        varAttrs[key, iconv.OvalName()]['_FillValue'] = -999.
+        varAttrs[key, iconv.OvalName()]['coordinates'] = 'longitude latitude station_elevation'
+        varAttrs[key, iconv.OvalName()]['units'] = 'unitless'
+        varDict[key]['errKey'] = key, iconv.OerrName()
+        varAttrs[key, iconv.OerrName()]['_FillValue'] = -999.
+        varAttrs[key, iconv.OerrName()]['coordinates'] = 'longitude latitude station_elevation'
+        varAttrs[key, iconv.OerrName()]['units'] = 'unitless'
+        varDict[key]['qcKey'] = key, iconv.OqcName()
+        varAttrs[key, iconv.OqcName()]['_FillValue'] = -999
+        varAttrs[key, iconv.OqcName()]['coordinates'] = 'longitude latitude station_elevation'
+        varAttrs[key, iconv.OqcName()]['units'] = ''
```

Comment:
Please check the values being assigned to the units attributes. What we finally decided on is:

* SI units when appropriate as specified in the spreadsheet (eg, air_temperature -> 'K')
* For string variables, use an empty string (ie, '')
* For unit-less variables
    * Categorical variables (eg, PreQC) use "unitless"
    * Mathematical variables (eg, mixing ratio)
        * If hidden units (eg, mixing ratio) express them - eg, "kg kg-1"
        * If pure numeric use "1"

The reason I'm asking is that it seems to me that AOD is a pure numeric quantity (unitless ratio with physical meaning) and would therefore have units set to '1'.

Thanks!